*** TkXpInterface.orig	Sun Jan 21 12:40:24 1996
--- TkXpInterface	Mon Aug  5 16:16:00 1996
***************
*** 313,319 ****
    entry $w.s.e -relief sunken -bd 2 -textvariable shutd \
          -fg $colact -bg $colbk -width 5
    checkbutton $w.s.lock -text "Lock the game" -anchor w \
!      -variable lock -relief flat -selector $colSe -fg $colact -bg $colbk
    bind $w.s.e <Return> " "
  
    pack $w.s.e $w.s.txt2 $w.s.lock -side left -anchor c
--- 313,319 ----
    entry $w.s.e -relief sunken -bd 2 -textvariable shutd \
          -fg $colact -bg $colbk -width 5
    checkbutton $w.s.lock -text "Lock the game" -anchor w \
!      -variable lock -relief flat  -fg $colact -bg $colbk
    bind $w.s.e <Return> " "
  
    pack $w.s.e $w.s.txt2 $w.s.lock -side left -anchor c
***************
*** 483,504 ****
      frame $w.frame.right -bg $colbk
      pack $w.frame.left $w.frame.right -side left -expand yes
      checkbutton $w.frame.left.s1v -text "$s1v Frames per second" -width 20 -anchor w \
!             -variable s1 -relief flat -selector $colSe -command "setsort $w s1" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s2v -text "$s2v Number of users" -width 20 -anchor w \
!             -variable s2 -relief flat -selector $colSe -command "setsort $w s2" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s3v -text "$s3v Map names" -width 20 -anchor w \
!             -variable s3 -relief flat -selector $colSe -command "setsort $w s3" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s4v -text "$s4v Version number" -width 20 -anchor w \
!             -variable s4 -relief flat -selector $colSe -command "setsort $w s4" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s5v -text "$s5v Countries" -width 20 -anchor w \
!             -variable s5 -relief flat -selector $colSe -command "setsort $w s5" -fg $colact -bg $colbk
  
      pack $w.frame.left.s1v $w.frame.left.s2v $w.frame.left.s3v $w.frame.left.s4v \
              $w.frame.left.s5v -side top -pady 2 -anchor w
      radiobutton $w.frame.right.b1 -fg $colact -bg $colbk -text "Ascending" \
!             -variable metsens -relief flat -value 0 -selector $colSe
      radiobutton $w.frame.right.b2 -fg $colact -bg $colbk -text "Descending" \
!             -variable metsens -relief flat -value 1 -selector $colSe
      pack $w.frame.right.b1 $w.frame.right.b2 \
           -side top -pady 2 -anchor w
  
--- 483,504 ----
      frame $w.frame.right -bg $colbk
      pack $w.frame.left $w.frame.right -side left -expand yes
      checkbutton $w.frame.left.s1v -text "$s1v Frames per second" -width 20 -anchor w \
!             -variable s1 -relief flat   -command "setsort $w s1" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s2v -text "$s2v Number of users" -width 20 -anchor w \
!             -variable s2 -relief flat   -command "setsort $w s2" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s3v -text "$s3v Map names" -width 20 -anchor w \
!             -variable s3 -relief flat   -command "setsort $w s3" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s4v -text "$s4v Version number" -width 20 -anchor w \
!             -variable s4 -relief flat   -command "setsort $w s4" -fg $colact -bg $colbk
      checkbutton $w.frame.left.s5v -text "$s5v Countries" -width 20 -anchor w \
!             -variable s5 -relief flat   -command "setsort $w s5" -fg $colact -bg $colbk
  
      pack $w.frame.left.s1v $w.frame.left.s2v $w.frame.left.s3v $w.frame.left.s4v \
              $w.frame.left.s5v -side top -pady 2 -anchor w
      radiobutton $w.frame.right.b1 -fg $colact -bg $colbk -text "Ascending" \
!             -variable metsens -relief flat -value 0  
      radiobutton $w.frame.right.b2 -fg $colact -bg $colbk -text "Descending" \
!             -variable metsens -relief flat -value 1  
      pack $w.frame.right.b1 $w.frame.right.b2 \
           -side top -pady 2 -anchor w
  
***************
*** 768,775 ****
  
      catch {destroy $w.cc}
      frame $w.cc -bg $colbk
!     canvas $w.cc.c -yscroll "$w.cc.vscroll set" -scrollincrement 20 -bg $colbk
!     scrollbar $w.cc.vscroll  -relief sunken -command "$w.cc.c yview" -fg $colact -bg $colbk
      $w.cc.c bind all <Any-Enter> "ChoosenLocal $w.cc.c $wor"
      $w.cc.c bind all <Button1-ButtonRelease> "set_local $w "
  
--- 768,775 ----
  
      catch {destroy $w.cc}
      frame $w.cc -bg $colbk
!     canvas $w.cc.c -yscrollcommand "$w.cc.vscroll set" -y -bg $colbk
!     scrollbar $w.cc.vscroll  -relief sunken -command "$w.cc.c yview" -bg $colbk
      $w.cc.c bind all <Any-Enter> "ChoosenLocal $w.cc.c $wor"
      $w.cc.c bind all <Button1-ButtonRelease> "set_local $w "
  
***************
*** 842,849 ****
  
      catch {destroy $w.cc}
      frame $w.cc -bg $colbk
!     canvas $w.cc.c -yscroll "$w.cc.vscroll set" -scrollincrement 20 -bg $colbk
!     scrollbar $w.cc.vscroll  -relief sunken -command "$w.cc.c yview" -fg $colact -bg $colbk
      $w.cc.c bind all <Any-Enter> "ChoosenMeta $w.cc.c $wor"
      $w.cc.c bind all <Button1-ButtonRelease> "set_remote $w "
  
--- 842,849 ----
  
      catch {destroy $w.cc}
      frame $w.cc -bg $colbk
!     canvas $w.cc.c -yscrollcommand "$w.cc.vscroll set" -yscrollincrement 20 -bg $colbk
!     scrollbar $w.cc.vscroll  -relief sunken -command "$w.cc.c yview" -bg $colbk
      $w.cc.c bind all <Any-Enter> "ChoosenMeta $w.cc.c $wor"
      $w.cc.c bind all <Button1-ButtonRelease> "set_remote $w "
  
***************
*** 1284,1291 ****
     frame $w.mp -bg $colbk
     frame $w.mp.1 -bg $colbk
     checkbutton $w.mp.1.but -text "Check options" -variable chkopt \
!      -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
!    radiobutton $w.mp.1.rad -relief flat -selector $colSe -fg $colact -bg $colbk
     pack  $w.mp.1.but $w.mp.1.rad -side left -expand yes -fill both
     pack  $w.mp.1 -side bottom -expand yes -fill both
     $w.mp.1.rad configure -state disabled
--- 1284,1291 ----
     frame $w.mp -bg $colbk
     frame $w.mp.1 -bg $colbk
     checkbutton $w.mp.1.but -text "Check options" -variable chkopt \
!      -anchor w -relief flat   -fg $colact -bg $colbk
!    radiobutton $w.mp.1.rad -relief flat   -fg $colact -bg $colbk
     pack  $w.mp.1.but $w.mp.1.rad -side left -expand yes -fill both
     pack  $w.mp.1 -side bottom -expand yes -fill both
     $w.mp.1.rad configure -state disabled
***************
*** 1293,1302 ****
  
      frame $w.mp.2 -bg $colbk
      scrollbar $w.mp.2.scroll  -relief sunken -command "$w.mp.2.list yview" \
!               -fg $colact -bg $colbk
!     listbox $w.mp.2.list -yscroll "$w.mp.2.scroll set" \
!             -relief sunken -geometry 17x21 -fg $colact -bg $colbk
!     tk_listboxSingleSelect $w.mp.2.list
      label $w.mp.2.msgt -text " Maps" -relief ridge -fg $colact -bg $colbk
      pack  $w.mp.2.msgt -side top -expand yes -fill both
  
--- 1293,1302 ----
  
      frame $w.mp.2 -bg $colbk
      scrollbar $w.mp.2.scroll  -relief sunken -command "$w.mp.2.list yview" \
!                -bg $colbk
!     listbox $w.mp.2.list -yscrollcommand "$w.mp.2.scroll set" \
!             -relief sunken -fg $colact -bg $colbk
! #    tk_listboxSingleSelect $w.mp.2.list
      label $w.mp.2.msgt -text " Maps" -relief ridge -fg $colact -bg $colbk
      pack  $w.mp.2.msgt -side top -expand yes -fill both
  
***************
*** 1320,1328 ****
    pack $w.def -side bottom -fill both -expand no
  
    pack $w.ship -side bottom -fill both -expand yes
!   canvas $c -width 100 -yscroll "$w.ship.vscroll set" \
!          -scrollincrement 100 -relief sunken -bg $colbk
!   scrollbar $w.ship.vscroll  -relief sunken -command "$c yview" -fg $colact -bg $colbk
    pack $w.ship.vscroll -side right -fill y
    pack $c -in $w.ship -expand yes -fill both
  
--- 1320,1328 ----
    pack $w.def -side bottom -fill both -expand no
  
    pack $w.ship -side bottom -fill both -expand yes
!   canvas $c -width 100  -relief sunken -bg $colbk -xscrollincrement 100 -yscrollincrement 100 \
!      -yscrollcommand "$w.ship.vscroll set"
!   scrollbar $w.ship.vscroll  -relief sunken -command "$c yview"  -bg $colbk
    pack $w.ship.vscroll -side right -fill y
    pack $c -in $w.ship -expand yes -fill both
  
***************
*** 1375,1392 ****
  #colG = foreground color of the polygonal shapes drawing
  #colR = foreground color of the other oval thrust shapes drawing
  
!    if {[tk colormodel .] == "color"} {# For color displays
        set colY "yellow"; set colBl "blue"
        set colG "green" ; set colR  "red"
!    } else {# If monchrome screen !
!       if $rev {
!          set colY "white" ; set colBl "white"
!          set colG "white" ; set colR  "white"
!       } else {
!          set colY "black" ; set colBl "black"
!          set colG "black" ; set colR  "black"
!       }
!    }
  
    set comto 0
    foreach shape $coms {incr comto 1}
--- 1375,1392 ----
  #colG = foreground color of the polygonal shapes drawing
  #colR = foreground color of the other oval thrust shapes drawing
  
! #   if {[tk colormodel .] == "color"} {# For color displays
        set colY "yellow"; set colBl "blue"
        set colG "green" ; set colR  "red"
! #   } else {# If monchrome screen !
! #      if $rev {
! #         set colY "white" ; set colBl "white"
! #         set colG "white" ; set colR  "white"
! #      } else {
! #         set colY "black" ; set colBl "black"
! #         set colG "black" ; set colR  "black"
! #      }
! # }
  
    set comto 0
    foreach shape $coms {incr comto 1}
***************
*** 2220,2243 ****
             break
           }
         }
!        if {$index != {} } {
!              $w select from $index
!              $w select to $index
!              $w yview [expr $index-12]
!        } else {
!            set item "$item*"
!            for {set i 0; set index {} ; set size [$w size] } { $i < $size } { incr i} {
!               if { [$w get $i] == $item } {
!                set index $i
!                break
!               }
!            }
!            if {$index != {} } {
!              $w select from $index
!              $w select to $index
!              $w yview [expr $index-12]
!            }
!        }
         catch {unset index i size item w}
  }
  ##################################
--- 2220,2243 ----
             break
           }
         }
! #        if {$index != {} } {
! #              $w select from $index
! #              $w select to $index
! #              $w yview [expr $index-12]
! #        } else {
! #            set item "$item*"
! #            for {set i 0; set index {} ; set size [$w size] } { $i < $size } { incr i} {
! #               if { [$w get $i] == $item } {
! #                set index $i
! #                break
! #               }
! #            }
! #            if {$index != {} } {
! #              $w select from $index
! #              $w select to $index
! #              $w yview [expr $index-12]
! #            }
! #        }
         catch {unset index i size item w}
  }
  ##################################
***************
*** 2402,2419 ****
      frame $g.b -bg $colbk
      checkbutton $g.b.gravityClockwise -text "Clockwise" -variable gravityClockwise \
       -anchor w -relief flat -command "gravity_check gravityClockwise" \
!      -selector $colSe -fg $colact -bg $colbk
  
      checkbutton $g.b.gravityAnticlockwise -text "Anticlockwise" \
!         -variable gravityAnticlockwise -selector $colSe -fg $colact -bg $colbk \
          -anchor w -relief flat -command "gravity_check gravityAnticlockwise"
  
      frame $g.l -bg $colbk
      checkbutton $g.l.gravityLine -text "Line" -variable gravityLine -fg $colact -bg $colbk\
!         -anchor w -relief flat -command "gravity_check gravityLine" -selector $colSe
  
      checkbutton $g.l.gravityPointSource -text "PointSource" -variable gravityPointSource -fg $colact -bg $colbk\
!         -anchor w -relief flat -command "gravity_check gravityPointSource" -selector $colSe
      pack $g.b.gravityClockwise $g.b.gravityAnticlockwise \
          -side left -expand yes -fill both
      pack $g.l.gravityPointSource $g.l.gravityLine \
--- 2402,2419 ----
      frame $g.b -bg $colbk
      checkbutton $g.b.gravityClockwise -text "Clockwise" -variable gravityClockwise \
       -anchor w -relief flat -command "gravity_check gravityClockwise" \
!        -fg $colact -bg $colbk
  
      checkbutton $g.b.gravityAnticlockwise -text "Anticlockwise" \
!         -variable gravityAnticlockwise   -fg $colact -bg $colbk \
          -anchor w -relief flat -command "gravity_check gravityAnticlockwise"
  
      frame $g.l -bg $colbk
      checkbutton $g.l.gravityLine -text "Line" -variable gravityLine -fg $colact -bg $colbk\
!         -anchor w -relief flat -command "gravity_check gravityLine"  
  
      checkbutton $g.l.gravityPointSource -text "PointSource" -variable gravityPointSource -fg $colact -bg $colbk\
!         -anchor w -relief flat -command "gravity_check gravityPointSource"  
      pack $g.b.gravityClockwise $g.b.gravityAnticlockwise \
          -side left -expand yes -fill both
      pack $g.l.gravityPointSource $g.l.gravityLine \
***************
*** 2594,2629 ****
  
      frame $w -bg $colbk
      checkbutton $w.noQuit -text "Server keeps running" -variable noQuit \
!         -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
    if {$state != "U"} {
      checkbutton $w.teamPlay -text "Team play mode" -variable teamPlay -anchor w \
!        -relief flat -command " maj_buttons teamPlay " -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
!     checkbutton $w.onePlayerOnly -text "One player only" -selector $colSe -variable onePlayerOnly\
         -anchor w -relief flat -command "maj_buttons onePlayerOnly" -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
      checkbutton $w.reportToMetaServer -text "Inform Meta-server" -fg $colact -bg $colbk \
!         -variable reportToMetaServer -anchor w -relief flat -selector $colSe
    }
  
    if {$state != "U"} {
      checkbutton $w.motdb -text "Use the server motd" -fg $colact -bg $colbk \
!         -variable motdb -anchor w -relief flat -selector $colSe
    }
  
    if {$state != "U"} {
      checkbutton $w.scoreb -text "Use the score file" -fg $colact -bg $colbk \
!         -variable scoreb -anchor w -relief flat -selector $colSe
    }
  
    if {$state != "U"} {
      checkbutton $w.timing -text "Race mode" -variable timing \
!         -anchor w -relief flat -command " maj_buttons timing " -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
--- 2594,2629 ----
  
      frame $w -bg $colbk
      checkbutton $w.noQuit -text "Server keeps running" -variable noQuit \
!         -anchor w -relief flat   -fg $colact -bg $colbk
  
    if {$state != "U"} {
      checkbutton $w.teamPlay -text "Team play mode" -variable teamPlay -anchor w \
!        -relief flat -command " maj_buttons teamPlay "   -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
!     checkbutton $w.onePlayerOnly -text "One player only"   -variable onePlayerOnly\
         -anchor w -relief flat -command "maj_buttons onePlayerOnly" -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
      checkbutton $w.reportToMetaServer -text "Inform Meta-server" -fg $colact -bg $colbk \
!         -variable reportToMetaServer -anchor w -relief flat  
    }
  
    if {$state != "U"} {
      checkbutton $w.motdb -text "Use the server motd" -fg $colact -bg $colbk \
!         -variable motdb -anchor w -relief flat  
    }
  
    if {$state != "U"} {
      checkbutton $w.scoreb -text "Use the score file" -fg $colact -bg $colbk \
!         -variable scoreb -anchor w -relief flat  
    }
  
    if {$state != "U"} {
      checkbutton $w.timing -text "Race mode" -variable timing \
!         -anchor w -relief flat -command " maj_buttons timing "   -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
***************
*** 2673,2678 ****
--- 2673,2679 ----
      pack $w.0.contactPort.entry $w.0.contactPort.label -side left
      bind $w.0.contactPort.entry <Return> " "
  
+ 
      if {$client == 3} {
    if {$state != "U"} {
         frame $w.0.team -bg $colbk
***************
*** 2729,2738 ****
      label $w.m.text3 -text " Size : $mapWidth x $mapHeight" -anchor w -fg $colact -bg $colbk
  
      checkbutton $w.m.dumpall -text "Save all options" -variable dumpall \
!        -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
      checkbutton $w.m.comprw -text "Compressed map file" -variable comprw \
!        -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
      pack $w.m.text1 $w.m.text2 $w.m.text3 $w.m.comprw $w.m.dumpall \
           -side top -fill both -pady 10 -padx 10
--- 2730,2739 ----
      label $w.m.text3 -text " Size : $mapWidth x $mapHeight" -anchor w -fg $colact -bg $colbk
  
      checkbutton $w.m.dumpall -text "Save all options" -variable dumpall \
!        -anchor w -relief flat   -fg $colact -bg $colbk
  
      checkbutton $w.m.comprw -text "Compressed map file" -variable comprw \
!        -anchor w -relief flat   -fg $colact -bg $colbk
  
      pack $w.m.text1 $w.m.text2 $w.m.text3 $w.m.comprw $w.m.dumpall \
           -side top -fill both -pady 10 -padx 10
***************
*** 2961,2973 ****
      label $w.msg.choix -text " Which file do you want for shipShape:" \
            -anchor c -fg $colact -bg $colbk
      frame $w.msg.cho -bg $colbk
!     checkbutton $w.msg.cho.shp -text "TkXpI.shp" -selector $colSe -fg $colact -bg $colbk \
          -variable shp -anchor w -relief flat -command "Verif_Choix $w.msg.cho shp "
!     checkbutton $w.msg.cho.xpi -text ".xpilotrc" -selector $colSe -fg $colact -bg $colbk \
          -variable xpi -anchor w -relief flat -command "Verif_Choix $w.msg.cho xpi"
!     checkbutton $w.msg.cho.bot -text "Both" -selector $colSe -fg $colact -bg $colbk \
          -variable bot -anchor w -relief flat -command "Verif_Choix $w.msg.cho bot"
!     checkbutton $w.msg.cho.noo -text "No one" -selector $colSe -fg $colact -bg $colbk \
          -variable noo -anchor w -relief flat -command "Verif_Choix $w.msg.cho noo"
      pack $w.msg.cho.shp $w.msg.cho.xpi $w.msg.cho.bot $w.msg.cho.noo -side left -fill both -expand yes
  
--- 2962,2974 ----
      label $w.msg.choix -text " Which file do you want for shipShape:" \
            -anchor c -fg $colact -bg $colbk
      frame $w.msg.cho -bg $colbk
!     checkbutton $w.msg.cho.shp -text "TkXpI.shp"   -fg $colact -bg $colbk \
          -variable shp -anchor w -relief flat -command "Verif_Choix $w.msg.cho shp "
!     checkbutton $w.msg.cho.xpi -text ".xpilotrc"   -fg $colact -bg $colbk \
          -variable xpi -anchor w -relief flat -command "Verif_Choix $w.msg.cho xpi"
!     checkbutton $w.msg.cho.bot -text "Both"   -fg $colact -bg $colbk \
          -variable bot -anchor w -relief flat -command "Verif_Choix $w.msg.cho bot"
!     checkbutton $w.msg.cho.noo -text "No one"   -fg $colact -bg $colbk \
          -variable noo -anchor w -relief flat -command "Verif_Choix $w.msg.cho noo"
      pack $w.msg.cho.shp $w.msg.cho.xpi $w.msg.cho.bot $w.msg.cho.noo -side left -fill both -expand yes
  
***************
*** 3009,3022 ****
      wm title $w "Robots"
      wm iconname $w "Robots"
  
!     checkbutton $w.robotsLeave -text "Robots leave the game" -selector $colSe -fg $colact -bg $colbk \
          -variable robotsLeave -anchor w -relief flat -command "Robots_leaving $w "
  
      checkbutton $w.idleRun -text "Robots keep playing" -variable idleRun \
! 	-anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
      checkbutton $w.robotsTalk -text "Robots talk" -variable robotsTalk \
!         -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
      frame $w.robotLeaveLife -bg $colbk
      label $w.robotLeaveLife.txt -text "Max life per robot" -fg $colact -bg $colbk
--- 3010,3023 ----
      wm title $w "Robots"
      wm iconname $w "Robots"
  
!     checkbutton $w.robotsLeave -text "Robots leave the game"   -fg $colact -bg $colbk \
          -variable robotsLeave -anchor w -relief flat -command "Robots_leaving $w "
  
      checkbutton $w.idleRun -text "Robots keep playing" -variable idleRun \
! 	-anchor w -relief flat   -fg $colact -bg $colbk
  
      checkbutton $w.robotsTalk -text "Robots talk" -variable robotsTalk \
!         -anchor w -relief flat   -fg $colact -bg $colbk
  
      frame $w.robotLeaveLife -bg $colbk
      label $w.robotLeaveLife.txt -text "Max life per robot" -fg $colact -bg $colbk
***************
*** 3083,3089 ****
              set v2 [string range $i [expr $pos+1] end]
              upvar #0 $v1 j
              checkbutton $w.$v1 -text $v2  -variable $v1 -fg $colact -bg $colbk \
!                       -anchor w -relief flat -selector $colSe
              pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
            }
       }
--- 3084,3090 ----
              set v2 [string range $i [expr $pos+1] end]
              upvar #0 $v1 j
              checkbutton $w.$v1 -text $v2  -variable $v1 -fg $colact -bg $colbk \
!                       -anchor w -relief flat  
              pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
            }
       }
***************
*** 3102,3120 ****
  
    if {$state != "U"} {
      checkbutton $w.playerStartsShielded -text "Start shielded" -variable \
!             playerStartsShielded -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
      checkbutton $w.allowShields -text "Use shields" -variable allowShields \
!        -anchor w -relief flat -command "no_shield $w" -selector $colSe -fg $colact -bg $colbk
    }
  
      checkbutton $w.shieldedItemPickup -text "Pick up with shields on" \
!         -variable shieldedItemPickup -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
      checkbutton $w.shieldedMining -text "Drop/throw mine with shields on" \
!         -variable shieldedMining -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
  
    if {$state != "U"} {
      pack $w.playerStartsShielded $w.allowShields $w.shieldedItemPickup \
--- 3103,3121 ----
  
    if {$state != "U"} {
      checkbutton $w.playerStartsShielded -text "Start shielded" -variable \
!             playerStartsShielded -anchor w -relief flat   -fg $colact -bg $colbk
    }
  
    if {$state != "U"} {
      checkbutton $w.allowShields -text "Use shields" -variable allowShields \
!        -anchor w -relief flat -command "no_shield $w"   -fg $colact -bg $colbk
    }
  
      checkbutton $w.shieldedItemPickup -text "Pick up with shields on" \
!         -variable shieldedItemPickup -anchor w -relief flat   -fg $colact -bg $colbk
  
      checkbutton $w.shieldedMining -text "Drop/throw mine with shields on" \
!         -variable shieldedMining -anchor w -relief flat   -fg $colact -bg $colbk
  
    if {$state != "U"} {
      pack $w.playerStartsShielded $w.allowShields $w.shieldedItemPickup \
***************
*** 3154,3163 ****
            upvar #0 $v1 j
            if { $v1=="playersOnRadar" } {
               checkbutton $w.$v1 -text $v2  -variable $v1 -anchor w -relief flat\
!              -command "no_visibility $w" -selector $colSe -fg $colact -bg $colbk
            } else {
              checkbutton $w.$v1 -text $v2  -variable $v1 \
!                   -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
            }
            pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
       }
--- 3155,3164 ----
            upvar #0 $v1 j
            if { $v1=="playersOnRadar" } {
               checkbutton $w.$v1 -text $v2  -variable $v1 -anchor w -relief flat\
!              -command "no_visibility $w"   -fg $colact -bg $colbk
            } else {
              checkbutton $w.$v1 -text $v2  -variable $v1 \
!                   -anchor w -relief flat   -fg $colact -bg $colbk
            }
            pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
       }
***************
*** 3232,3238 ****
  	if {($state == "U") && ($k != 1) && ($k != 5) } {
          } else {
            checkbutton $w.$v1 -text $v2  -variable $v1 -anchor w -relief flat \
!                       -selector $colSe -fg $colact -bg $colbk
            pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
          }
       }
--- 3233,3239 ----
  	if {($state == "U") && ($k != 1) && ($k != 5) } {
          } else {
            checkbutton $w.$v1 -text $v2  -variable $v1 -anchor w -relief flat \
!                         -fg $colact -bg $colbk
            pack $w.$v1 -side top -fill both -expand yes -padx 10 -pady 5
          }
       }
***************
*** 3276,3298 ****
      wm iconname $w "Player"
  
    if {$state != "U" } {
!     checkbutton $w.allowPlayerCrashes -text "Players can crash into each other"\
!         -variable allowPlayerCrashes -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.allowPlayerKilling -text "Players can kill each other" \
!         -variable allowPlayerKilling -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.allowPlayerBounces -text "Players bounce with other players" \
!         -variable allowPlayerBounces -anchor w -relief flat -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.limitedLives -text "Limited lives" -variable limitedLives \
!         -anchor w -relief flat -command "lim_worldLives $w " -selector $colSe -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
--- 3277,3299 ----
      wm iconname $w "Player"
  
    if {$state != "U" } {
!     checkbutton $w.allowPlayerCrashes -text "Players can crash into each other" \
!         -variable allowPlayerCrashes -anchor w -relief flat   -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.allowPlayerKilling -text "Players can kill each other" \
!         -variable allowPlayerKilling -anchor w -relief flat   -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.allowPlayerBounces -text "Players bounce with other players" \
!         -variable allowPlayerBounces -anchor w -relief flat   -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
      checkbutton $w.limitedLives -text "Limited lives" -variable limitedLives \
!         -anchor w -relief flat -command "lim_worldLives $w "   -fg $colact -bg $colbk
    }
  
    if {$state != "U" } {
***************
*** 3334,3344 ****
    if {$state != "U"} {
     frame $w.e -bg $colbk
     checkbutton $w.e.edgeWrap -text "Edge wrap" -variable edgeWrap -anchor w \
!         -relief flat -command "edges $w edgeWrap" -selector $colSe -fg $colact -bg $colbk
     checkbutton $w.e.edgeBounce -text "Edge bouncing" -variable edgeBounce \
!         -anchor w -relief flat -command "edges $w edgeBounce" -selector $colSe -fg $colact -bg $colbk
     checkbutton $w.e.extraBorder -text "Extra border" -variable extraBorder \
!         -anchor w -relief flat -command "edges $w extraBorder" -selector $colSe -fg $colact -bg $colbk
  
     pack $w.e.extraBorder $w.e.edgeWrap $w.e.edgeBounce \
         -side left -expand yes -fill both
--- 3335,3345 ----
    if {$state != "U"} {
     frame $w.e -bg $colbk
     checkbutton $w.e.edgeWrap -text "Edge wrap" -variable edgeWrap -anchor w \
!         -relief flat -command "edges $w edgeWrap"   -fg $colact -bg $colbk
     checkbutton $w.e.edgeBounce -text "Edge bouncing" -variable edgeBounce \
!         -anchor w -relief flat -command "edges $w edgeBounce"   -fg $colact -bg $colbk
     checkbutton $w.e.extraBorder -text "Extra border" -variable extraBorder \
!         -anchor w -relief flat -command "edges $w extraBorder"   -fg $colact -bg $colbk
  
     pack $w.e.extraBorder $w.e.edgeWrap $w.e.edgeBounce \
         -side left -expand yes -fill both
***************
*** 3358,3364 ****
         set v2 [string range $i [expr $pos+1] end]
         upvar #0 $v1 j
         checkbutton $w.b.$v1 -text $v2  -variable $v1 -anchor w -relief flat \
!                    -selector $colSe -fg $colact -bg $colbk
         pack $w.b.$v1 -side left -fill both -expand yes
     }
     pack $w.b -side top -expand yes -fill both -padx 10 -pady 5
--- 3359,3365 ----
         set v2 [string range $i [expr $pos+1] end]
         upvar #0 $v1 j
         checkbutton $w.b.$v1 -text $v2  -variable $v1 -anchor w -relief flat \
!                      -fg $colact -bg $colbk
         pack $w.b.$v1 -side left -fill both -expand yes
     }
     pack $w.b -side top -expand yes -fill both -padx 10 -pady 5
***************
*** 3437,3448 ****
       checkbutton $w.allowShipShapes \
         -text "Players allowed to define their own ship shape" \
         -variable allowShipShapes -anchor w -relief flat -padx 10 -pady 5 \
!        -selector $colSe -fg $colact -bg $colbk
    }
  
      checkbutton $w.shotsGravity -text "Are bullets affected by gravity" \
         -variable shotsGravity -anchor w -relief flat -pady 5 \
!        -selector $colSe -fg $colact -bg $colbk
       set k 0
       foreach i {shipMass=Ship\ mass \
          shotMass=Shot\ mass \
--- 3438,3449 ----
       checkbutton $w.allowShipShapes \
         -text "Players allowed to define their own ship shape" \
         -variable allowShipShapes -anchor w -relief flat -padx 10 -pady 5 \
!          -fg $colact -bg $colbk
    }
  
      checkbutton $w.shotsGravity -text "Are bullets affected by gravity" \
         -variable shotsGravity -anchor w -relief flat -pady 5 \
!          -fg $colact -bg $colbk
       set k 0
       foreach i {shipMass=Ship\ mass \
          shotMass=Shot\ mass \
***************
*** 3563,3569 ****
  
      button $w.ok -text Quit -command "destroy $w" -fg $colact -bg $colbk
      text $w.t -relief raised -bd 2 -yscrollcommand "$w.s set" -fg $colact -bg $colbk
!     scrollbar $w.s -relief flat -command "$w.t yview" -fg $colact -bg $colbk
      pack $w.ok -side bottom -fill x -expand yes
      pack $w.s -side right -fill y
  
--- 3564,3570 ----
  
      button $w.ok -text Quit -command "destroy $w" -fg $colact -bg $colbk
      text $w.t -relief raised -bd 2 -yscrollcommand "$w.s set" -fg $colact -bg $colbk
!     scrollbar $w.s -relief flat -command "$w.t yview"  -bg $colbk
      pack $w.ok -side bottom -fill x -expand yes
      pack $w.s -side right -fill y
  
***************
*** 3858,3872 ****
  	}
   }
  
!  if {[tk colormodel .] != "color"} {# For monochrome displays
!    if $rev {
!     set colact "white" ; set coldes "white" ; set colsc "black"
!     set colSe  "white" ; set colbk  "black"
!    } else {
!     set colact "black" ; set coldes "black" ; set colsc "white"
!     set colSe  "black" ; set colbk  "white"
!    }
!  }
  
  set home_dir $env(HOME)
  set hostname 127.0.0.1
--- 3859,3873 ----
  	}
   }
  
! # if {[tk colormodel .] != "color"} {# For monochrome displays
! #   if $rev {
! #    set colact "white" ; set coldes "white" ; set colsc "black"
! #    set colSe  "white" ; set colbk  "black"
! #   } else {
! #    set colact "black" ; set coldes "black" ; set colsc "white"
! #    set colSe  "black" ; set colbk  "white"
! #   }
! # }
  
  set home_dir $env(HOME)
  set hostname 127.0.0.1
***************
*** 3919,3922 ****
       showcw .botbuttons.copyb 2 Root
       mainproc 0
   }
! ######################################  The End #####################################
--- 3920,3923 ----
       showcw .botbuttons.copyb 2 Root
       mainproc 0
   }
! ################################  The End #####################################

